plugins/defaultplugins/ifplugin/interface_config_test.go

```diff
@@ -0,0 +1,302 @@
+
+package ifplugin
+
+import (
+	"net"
+	"testing"
+	"time"
+
+	"git.fd.io/govpp.git/adapter/mock"
+	govppapi "git.fd.io/govpp.git/api"
+	"git.fd.io/govpp.git/core"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/vpp-agent/idxvpp/nametoidx"
+	dhcp_api "github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/dhcp"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/interfaces"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/ip"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/memif"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/tap"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/tapv2"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/vpe"
+	if_api "github.com/ligato/vpp-agent/plugins/defaultplugins/common/model/interfaces"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/ifaceidx"
+	"github.com/ligato/vpp-agent/tests/vppcallmock"
+	. "github.com/onsi/gomega"
+)
+
+var ifNames = []string{"if1", "if2", "if3"}
+var mtu uint32 = 1500
+var netAddresses = []string{"10.0.0.1/24", "10.0.0.2/24", "192.168.50.1/24"}
+var ipv6Addresses = []string{"fd21:7408:186f::/48", "2001:db8:a0b:12f0::1/48"}
+var macs = []string{"46:06:18:DB:05:3A", "BC:FE:E9:5E:07:04"}
+
+/* Interface configurator init and close */
+
+// Test init function
+func TestInterfaceConfiguratorInit(t *testing.T) {
+	var err error
+	// Setup
+	RegisterTestingT(t)
+	ctx := &vppcallmock.TestCtx{
+		MockVpp: &mock.VppAdapter{},
+	}
+	connection, _ := core.Connect(ctx.MockVpp)
+	plugin := &InterfaceConfigurator{
+		Log:      logrus.DefaultLogger(),
+		GoVppmux: connection,
+	}
+	swIfIndices := ifaceidx.NewSwIfIndex(nametoidx.NewNameToIdx(plugin.Log, "swIf-test", "swIf", nil))
+	dhcpIndices := ifaceidx.NewDHCPIndex(nametoidx.NewNameToIdx(plugin.Log, "dhcp-test", "dhcp", nil))
+	ifVppNotifChan := make(chan govppapi.Message, 100)
+	// Reply set
+	ctx.MockVpp.MockReply(&memif.MemifSocketFilenameDetails{
+		SocketID:       1,
+		SocketFilename: []byte("test-socket-filename"),
+	})
+	ctx.MockVpp.MockReply(&vpe.ControlPingReply{})
+	// Register
+	swIfIndices.RegisterName(ifNames[0], 1, nil)
+	swIfIndices.RegisterName(ifNames[1], 2, nil)
+	// Test init
+	err = plugin.Init(swIfIndices, dhcpIndices, mtu, ifVppNotifChan)
+	Expect(err).To(BeNil())
+	Expect(plugin.swIfIndexes).ToNot(BeNil())
+	Expect(plugin.dhcpIndices).ToNot(BeNil())
+	Expect(plugin.notifChan).ToNot(BeNil())
+	Expect(plugin.mtu).To(BeEquivalentTo(mtu))
+	Expect(plugin.afPacketConfigurator).ToNot(BeNil())
+	Expect(plugin.memifScCache["test-socket-filename"]).To(BeEquivalentTo(1))
+	// Test DHCP notifications
+	dhcpIpv4 := &dhcp_api.DhcpComplEvent{
+		HostAddress:   net.ParseIP(ipAddresses[0]),
+		RouterAddress: net.ParseIP(ipAddresses[1]),
+		HostMac: func(mac string) []byte {
+			parsed, err := net.ParseMAC(mac)
+			Expect(err).To(BeNil())
```

Comment:
What is actually being tested here? Just set `HostMac` as `[]byte{0xAA, 0xBB...etc}`.

Reply:
parseMac is used widely across the plugins so I think it is not a bad idea to put the check into tests even if it is not directly related to the test case

Reply:
But you are not testing ParseMAC from any test here, you are using it independently here. ParseMAC as a function is tested in net package.